### ISSUE TITLE
[Crash] Attempt to invoke virtual method 'void java.io.InputStream.close()' on a null object reference

### ISSUE THREAD

**Issue by AndrewJack** [commonsguy/cwac-netsecurity#3]
**Android OS:** 4.4.4, 5.1, 6.0
**Devices:** Sony Xperia Z1, Z2, and Z3

**To reproduce:**
Run the demo app on one of the above devices, app will crash on startup with the following stacktrace.

Potential fix: 
Surround `is.close()` with null check. [DirectoryCertificateSource.java - Line 199](https://github.com/commonsguy/cwac-netsecurity/blob/f17cfe9c9b18816e18f375e6e4b47bba76db26a6/netsecurity/src/main/java/com/commonsware/cwac/netsecurity/config/DirectoryCertificateSource.java#L199)

```
java.lang.RuntimeException: Unable to create application com.example.MainApplication: java.lang.NullPointerException: Attempt to invoke virtual method 'void java.io.InputStream.close()' on a null object reference
	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:4809)
	at android.app.ActivityThread.access$1600(ActivityThread.java:154)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1452)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:234)
	at android.app.ActivityThread.main(ActivityThread.java:5526)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void java.io.InputStream.close()' on a null object reference
	at com.commonsware.cwac.netsecurity.config.DirectoryCertificateSource.readCertificate(DirectoryCertificateSource.java:199)
	at com.commonsware.cwac.netsecurity.config.DirectoryCertificateSource.getCertificates(DirectoryCertificateSource.java:77)
	at com.commonsware.cwac.netsecurity.config.SystemCertificateSource.getCertificates(SystemCertificateSource.java:27)
	at com.commonsware.cwac.netsecurity.config.CertificatesEntryRef.getTrustAnchors(CertificatesEntryRef.java:40)
	at com.commonsware.cwac.netsecurity.config.NetworkSecurityConfig.getTrustAnchors(NetworkSecurityConfig.java:86)
	at com.commonsware.cwac.netsecurity.config.NetworkSecurityTrustManager.getAcceptedIssuers(NetworkSecurityTrustManager.java:171)
	at com.commonsware.cwac.netsecurity.config.RootTrustManager.getAcceptedIssuers(RootTrustManager.java:170)
	at com.commonsware.cwac.netsecurity.X509ExtensionsWrapper.getAcceptedIssuers(X509ExtensionsWrapper.java:147)
	at com.commonsware.cwac.netsecurity.CompositeTrustManager.getAcceptedIssuers(CompositeTrustManager.java:301)
	at okhttp3.internal.tls.TrustRootIndex.get(TrustRootIndex.java:43)
	at okhttp3.internal.platform.Platform.buildCertificateChainCleaner(Platform.java:147)
	at okhttp3.internal.platform.AndroidPlatform.buildCertificateChainCleaner(AndroidPlatform.java:159)
	at okhttp3.internal.tls.CertificateChainCleaner.get(CertificateChainCleaner.java:41)
	at okhttp3.OkHttpClient$Builder.sslSocketFactory(OkHttpClient.java:641)
	at com.commonsware.cwac.netsecurity.OkHttp3Integrator.applyTo(OkHttp3Integrator.java:35)
	at com.example.MainApplication.applyNetSecurityConfig(MainApplication.java:103)
	at com.example.MainApplication.onCreate(MainApplication.java:59)
```

**Comment by commonsguy** [comment]
In the AOSP code, this stuff uses `IoUtils.closeQuietly()`, which is internal to Android. I botched the replacement, as I should be doing this sort of `null` check. I merged your PR (thanks!), and I'll work on fixing the other occurrences of this, as there will be quite a few of them...

**Comment by commonsguy** [comment]
Reopening until I release a new artifact that contains these fixes.

**Comment by commonsguy** [comment]
Addressed in v0.2.1.
